### PR TITLE
Add Host.Always log category

### LINKED ISSUE #6252: Log when host/worker is ready for VS Code debugger to attach
After discussing with @pragnagopa, it sounds like the initial change in #6230 is not quite sufficient for the purposes of VS Code attaching the debugger reliably.

There are 2 issues in particular:
- In-proc .NET apps will not print the new message
- The message is printed with log level Information. If default log levels are changed to Warning, Error, or None, the message will not print

We need this message to be printed regardless of stack or log level. Also suggest updating the message to something more generic and doesn't mention "worker process". (something like "Function app is ready"?)

Also needed for this: https://github.com/Azure/azure-functions-core-tools/issues/1131

@ericjizbaMSFT

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #6252

### CODE REVIEW

File: src/WebJobs.Script/Diagnostics/Extensions/ExtensionBundleLoggerExtension.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ internal static class ExtensionBundleLoggerExtension
 
         private static readonly Action<ILogger, string, Exception> _contentProviderNotConfigured =
             LoggerMessage.Define<string>(
-            LogLevel.Information,
```
Comment (pragnagopa) [inline src/WebJobs.Script/Diagnostics/Extensions/ExtensionBundleLoggerExtension.cs]: @soninaren - Changing logs from Extension bundles to debug level.  For system logs default log level is Debug. We will continue to see these logs on Kusto. Can you please ACK?

File: src/WebJobs.Script/Extensions/ScriptLoggingBuilderExtensions.cs [hunk 2]
```diff
@@ -14,21 +16,24 @@ public static class ScriptLoggingBuilderExtensions
...
             builder.AddFilter((c, l) => Filter(c, l, LogLevel.Information));
+            builder.AddFilter(ScriptConstants.LogCategoryAlways, LogLevel.Information);
         }
 
         public static void AddDefaultWebJobsFilters<T>(this ILoggingBuilder builder, LogLevel level) where T : ILoggerProvider
         {
             builder.AddFilter<T>(null, LogLevel.None);
-            builder.AddFilter<T>((c, l) => Filter(c, l, LogLevel.Trace));
+            builder.AddFilter<T>((c, l) => Filter(c, l, level));
+            builder.AddFilter<T>(ScriptConstants.LogCategoryAlways, LogLevel.Information);
```
Comment (pragnagopa) [inline src/WebJobs.Script/Extensions/ScriptLoggingBuilderExtensions.cs:36]: cc @anthonychu 

@brettsam  - Introduced new log category to ensure logs are always emitted. Let me know if there is a better way to achieve this.

File: src/WebJobs.Script/Diagnostics/Extensions/ExtensionBundleLoggerExtension.cs (hunk 1 above)
Comment (soninaren) [reply src/WebJobs.Script/Diagnostics/Extensions/ExtensionBundleLoggerExtension.cs]: Will these logs appear in the core tools? We should be good for this particular message.

File: src/WebJobs.Script/Diagnostics/Extensions/ExtensionBundleLoggerExtension.cs [hunk 3]
```diff
@@ -25,31 +25,31 @@ internal static class ExtensionBundleLoggerExtension
 
         private static readonly Action<ILogger, string, string, Exception> _locateExtensionBundle =
             LoggerMessage.Define<string, string>(
-            LogLevel.Information,
+            LogLevel.Debug,
             new EventId(102, nameof(LocateExtensionBundle)),
```
Comment (soninaren) [inline src/WebJobs.Script/Diagnostics/Extensions/ExtensionBundleLoggerExtension.cs]: It is important that this message to appear in core tools, We have used this to identify issues root cause in a number of GitHub issues.

Comment (pragnagopa) [reply src/WebJobs.Script/Diagnostics/Extensions/ExtensionBundleLoggerExtension.cs]: Default log level when running in Core Tools is Info. Log level needs to be set to debug to see these logs.

```json
"logging": {
    "logLevel": {
      "default": "Debug"
    }
  }
```

Comment (soninaren) [reply src/WebJobs.Script/Diagnostics/Extensions/ExtensionBundleLoggerExtension.cs]: I don't think we can mark all the extension bundle logs as Debug level. Would have to look at them individually.

Comment (pragnagopa) [reply src/WebJobs.Script/Diagnostics/Extensions/ExtensionBundleLoggerExtension.cs]: Ok. Reverted changes. Was trying to clean up info logs to debug :)

Comment (soninaren) [reply src/WebJobs.Script/Diagnostics/Extensions/ExtensionBundleLoggerExtension.cs]: I can review and mark extension bundle logs as debug. I think we only need a few of them to be info. Rest can be debug.

**Comment by brettsam** [comment]
This is only needed by Core Tools, right? Could we --
1. Have CoreTools wire up its own Logging Provider and add a filter there to make sure that this provider is always logged.
2. If this logger sees this `WorkerProcessInitialized` event, it writes it to the Console. It ignores everything else.

That'd potentially result in one duplicate log under normal circumstances, but it'd remove this new filter and category from the host.

**Comment by pragnagopa** [comment]
Thanks @brettsam !

Yes, this change is only needed for CoreTools and specifically for VSCode to figure out when is the right time to attach a debugger.

For out-of-proc, need to emit `WorkerProcessInitialized` log and for C# need to emit `Host lock lease acquired` even if `logLevel` is set to `None`. I will look into adding the filter in core tools and test it out.

**Comment by pragnagopa** [comment]
Closing in favor https://github.com/Azure/azure-functions-core-tools/pull/2139

**Pull request closed without merging**